presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -227,6 +236,17 @@ public TaskStatus process()
                     return TaskStatus.finished();
                 }
                 ListenableFuture<?> future;
+                /*
+                 * Block until one of below conditions is met:
+                 *  1. Arrival of first non "all" filter
```

Comment:
`all` filter should trigger DF completion (and filter unblock). Remove this condition

Reply:
Removed comment